src/robot_interface/robot_interface.py

```diff
@@ -1,146 +1,214 @@
-from robot_interface.models.geometry.pose import Pose
-from robot_interface.models.inspection.inspection import Inspection, InspectionResult
-from robot_interface.models.mission import MissionStatus, Step
+from robot_interface.models.inspection.inspection import InspectionResult
+from robot_interface.models.sensor.status import SensorStatus
+from robot_interface.models.step.step import InspectionStep, Step
+from robot_interface.models.step.status import StepStatus
+from robot_interface.models.sensor.sensor import Sensor
 
 
 class RobotInterface(metaclass=ABCMeta):
     """Interface to communicate with robots."""
 
     @abstractmethod
-    def schedule_step(self, step: Step) -> Tuple[bool, Optional[Any], Optional[Joints]]:
-        """Schedules a Mission on the robot.
+    def stop(self) -> None:
+        """Stops the execution of the current scheduled step
+        and stops the movements of the robot.
 
-        The method must adapt the standard mission to the
-        specific robots mission planning.
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        RobotException if function fails
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def dock(self) -> None:
+        """Orders to move back to the docking station and start
+        the docking procedure.
 
         Parameters
         ----------
-        step : Step
-            Mission object describing the mission.
+        None
 
         Returns
         -------
-        bool
-            True if successful of scheduling a mission, false otherwise.
+        None
+
+        Raises
+        ------
+        RobotException if function fails
         """
         raise NotImplementedError
 
     @abstractmethod
-    def mission_scheduled(self) -> bool:
-        """Determines if a mission is scheduled on the robot.
+    def schedule_step(self, step: Step) -> None:
+        """Schedules a step on the robot.
+
+        The method must adapt the standard step to the
+        specific robots mission planning.
 
-        If a mission is scheduled it may be that another mission
-        should not be scheduled.
+        Parameters
+        ----------
+        step : Step
+            Step object that describes the intended task for the
+            robot to perform.
 
         Returns
         -------
-        bool
-            True if a mission is already scheduled, false otherwise.
+        None
+
+        Raises
+        ------
+        RobotException if function fails
         """
         raise NotImplementedError
 
     @abstractmethod
-    def mission_status(self, mission_id: Any) -> MissionStatus:
-        """Retrieves status of the current executing mission for the robot.
+    def step_status(self, step: Optional[Step] = None) -> StepStatus:
+        """Function that returns the status of the progress for
+        a given step.
```

Comment:
Are we interested in checking the status of any step of a mission?

Reply:
I would think of this as checking the status of the current step in a mission. Otherwise we will have to implement some way of storing the state of previous steps.